code/game/machinery/_machines_base/machinery.dm

```diff
@@ -134,12 +134,20 @@ Class Procs:
 	RefreshParts()
 	power_change()
 
+	var/area/A = get_area(src)
+	if(A)
+		LAZYADD(A.machinery_list, src) // Add machinery object to area machinery list
```

Comment:
Would suggest using `weakref(src)` and then `foo.resolve()` in the loop over the list, to avoid GC problems if machinery isn't properly cleaned up from the area lists.